.github/workflows/flow.yml

```diff
@@ -131,18 +131,39 @@ jobs:
 
       - name: Mongo
         run: |
-          docker run -d --rm --name mongo -p 27017:27017 mongo:8.0.10 --replSet rs0 --bind_ip_all
+          echo "starting mongoDB..."
+          docker run -d --rm --name mongo -p 27017:27017 mongo:${{ matrix.db-version.mongo }} \
+            bash -c 'openssl rand -base64 756 > /data/mongo.key && chmod 400 /data/mongo.key && mongod --replSet rs0 --bind_ip_all --keyFile /data/mongo.key'
+
           until docker exec mongo mongosh --eval 'db.runCommand({ ping: 1 })' &> /dev/null; do
-            echo "Waiting for MongoDB to be ready..."
+            echo "waiting for MongoDB to be ready..."
             sleep 2
           done
+
+          echo "initialize replica set"
           docker exec mongo mongosh --eval 'rs.initiate({
             _id: "rs0",
-            members: [
-              { _id: 0, host: "localhost:27017" }
-            ]
+            members: [{ _id: 0, host: "localhost:27017" }]
           })'
 
+          echo "create admin user"
+          docker exec mongo mongosh --eval '
+            db = db.getSiblingDB("admin");
+            db.createUser({
+              user: "admin",
+              pwd: "admin",
+              roles: ["root"]
+            })'
+
+          echo "create non-admin user for tests"
+          docker exec mongo mongosh -u admin -p admin --eval '
+            db = db.getSiblingDB("admin");
+            db.createUser({
+              user: "ciuser",
+              pwd: "cipass",
+              roles: ["readWriteAnyDatabase"]
```

Comment:
this may be the less privileged role we need to make mongo connector work as expected. since we need `ListDatabases` privileges during discovery.